homeassistant/components/omnilogic/light.py

```diff
@@ -0,0 +1,225 @@
+"""Platform for light integration."""
+import logging
+import time
+
+from omnilogic import LightEffect, OmniLogicException
+import voluptuous as vol
+
+from homeassistant.components.light import ATTR_EFFECT, SUPPORT_EFFECT, LightEntity
+from homeassistant.helpers import entity_platform
+
+from .common import OmniLogicEntity, OmniLogicUpdateCoordinator
+from .const import COORDINATOR, DOMAIN
+
+SERVICE_SET_V2EFFECT = "set_v2_lights"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the light platform."""
+
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+    entities = []
+
+    for item_id, item in coordinator.data.items():
+        id_len = len(item_id)
+        item_kind = item_id[-2]
+        entity_settings = LIGHT_TYPES.get((id_len, item_kind))
+
+        if not entity_settings:
+            continue
+
+        for entity_setting in entity_settings:
+            for state_key, entity_class in entity_setting["entity_classes"].items():
+                if state_key not in item:
+                    continue
+
+                guard = False
+                for guard_condition in entity_setting["guard_condition"]:
+                    if guard_condition and all(
+                        item.get(guard_key) == guard_value
+                        for guard_key, guard_value in guard_condition.items()
+                    ):
+                        guard = True
+
+                if guard:
+                    continue
+
+                entity = entity_class(
+                    coordinator=coordinator,
+                    state_key=state_key,
+                    name=entity_setting["name"],
+                    kind=entity_setting["kind"],
+                    item_id=item_id,
+                    icon=entity_setting["icon"],
+                )
+
+                entities.append(entity)
+
+    async_add_entities(entities)
+
+    # register service
+    platform = entity_platform.current_platform.get()
+
+    platform.async_register_entity_service(
+        SERVICE_SET_V2EFFECT,
+        {
+            vol.Optional("brightness"): vol.All(
+                vol.Coerce(int), vol.Range(min=0, max=4)
+            ),
+            vol.Optional("speed"): vol.All(vol.Coerce(int), vol.Range(min=0, max=8)),
+        },
+        "async_set_v2effect",
+    )
+
+
+class OmniLogicLightControl(OmniLogicEntity, LightEntity):
+    """Define an Omnilogic Light entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        icon: str,
+        item_id: tuple,
+        state_key: str,
```

Comment:
Please consider using `EntityDescription`s instead.

Reply:
Not sure what you mean on this one.

Reply:
e.g. something like https://github.com/home-assistant/core/blob/f1979f8b68035f98bd493fb7c34b98176e7bedde/homeassistant/components/fjaraskupan/binary_sensor.py#L27 ?